modules/logging/README.md

```diff
@@ -0,0 +1,398 @@
+
+```
+┌─────────────────┐                 ┌─────────────────┐                 ┌─────────────────┐
+│  Thread 1       │                 │  Thread 2       │                 │  Thread 3       │
+│                 │                 │                 │                 │                 │
+│  ┌───────────┐  │                 │  ┌───────────┐  │                 │  ┌───────────┐  │
+│  │ThreadLocal│  │ Thread Switch   │  │ThreadLocal│  │ Thread Switch   │  │ThreadLocal│  │
+│  │   MDC     │──┼─-────X─────────>│  │   MDC     │──┼───-──X─────────>│  │   MDC     │  │
+│  └───────────┘  │                 │  └───────────┘  │                 │  └───────────┘  │
+└─────────────────┘                 └─────────────────┘                 └─────────────────┘
+        │                                   │                                   │
+        │                                   │                                   │
+        │         ┌───────────────────────────────────────────────┐             │
+        │         │                                               │             │
+        └────────>│              Reactor Context                  │<────────────┘
+                  │    (MDC data flows with the reactive chain)   │
+                  └───────────────────────────────────────────────┘
+```
+
+**How it works:**
+
+1. `MDCWebFilter` captures MDC data and stores it in Reactor Context via `contextWrite()`
+2. The context is attached to the reactive chain and flows with operations across thread boundaries
+3. `ReactorContextHolder` provides utilities to move data between thread-local MDC and Reactor Context
+4. When logging occurs, we temporarily restore the MDC data from Reactor Context to the current thread
+5. After logging, we clean up the thread-local MDC to prevent leaks
+
+### Access Logging
+
+The module provides a configurable access log filter that captures information about HTTP requests:
+
+- HTTP method, URI, status code, and processing duration
+- Configurable log levels (info, debug, trace) based on URI patterns
+- Integration with MDC for enriched context in logs
+
+### Structured JSON Logging
+
+Includes a logback configuration for JSON-formatted logs using the Logstash encoder:
+
+- Consistent structured logging format for easy parsing and analysis
+- Automatic inclusion of all MDC properties in logs
+- Configurable through standard logback properties
+
+### Spring Cloud Gateway Integration
+
+Special support for Spring Cloud Gateway:
+
+- Global filter adapters to integrate with the Gateway filter chain
+- Proper handling of access logs in the Gateway environment
+- MDC propagation throughout Gateway's reactive processing pipeline
+
+## Usage
+
+### Maven Dependency
+
+```xml
+<dependency>
+    <groupId>org.georchestra</groupId>
+    <artifactId>georchestra-gateway-logging</artifactId>
+    <version>${revision}</version>
+</dependency>
+```
```

Comment:
probably outside the scope of this PR, but it would be interesting to add a GHA which would publish the module onto geOrchestra's artifactory.

Reply:
yes, shut up
no, seriously, if you wanted to make a standalone library out of it we could also port the servlet support so we'd have a single library for servlet, webflux, and webflux + gateway